Display spaces

benmorrismayden opened the pull request.

ArcticFoxJ committed:
add basic scss file structure

benmorrismayden committed:
Create placeholder directory structure.

benmorrismayden committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ArcticFoxJ committed:
link style.css to index

ArcticFoxJ committed:
link bootstrap

ArcticFoxJ committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ArcticFoxJ committed:
removed duplicate file

LukeW20 committed:
initial setup of database

LukeW20 committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
added html structure

benmorrismayden committed:
Create Carpark class to query carpark database

aimeeew committed:
Added key colours as variables

aimeeew committed:
changed file structure

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
added sass-cache to gitignore

Commit:
create class autoloader

Commit:
change autoloader to include once

rmjjjt committed:
Added Basic Functionality For Dual Calendar Display.

Added and linked files for jquery-1.12.4.min.js, datepicker js and css files as well as calendar.js
as well as calendar.html.

rmjjjt committed:
Amended Calendar Cunctionality.

Changed datepicker.min.js output to Uk format DDMMYYYY.
Set from calendar to default to current month.
Set to calendar to default to next month.
Added multiple selection function to both calendars.
Added a clear button to remove all selections in each calendar.

rmjjjt committed:
Added Today Button and Start Date.

Start date stops user being able to view dates in the past.
Today button enables user to return to current month WITHOUT selecting today's date.

aimeeew committed:
Swapped logo font for image and edited background image

aimeeew committed:
Took out unnecessary bootstrap div

aimeeew committed:
Modified logo size

aimeeew committed:
Added page title

aimeeew committed:
Added background colours

aimeeew committed:
Refined html structure

aimeeew committed:
Created css file for main and added to imports

aimeeew committed:
Added button classes

aimeeew committed:
Rearranged cascading order

aimeeew committed:
Altered user input layout

aimeeew committed:
Positioned button - draft

aimeeew committed:
Altered default button styles

aimeeew committed:
Removed button classes

aimeeew committed:
Added border edge to logo box

aimeeew committed:
Enlargened footer

aimeeew committed:
Removed border on touching buttons

aimeeew committed:
Changed button colours

aimeeew committed:
Updated stylesheets

aimeeew committed:
Added box shadow to header

aimeeew committed:
Deleted previous watcher sheet

aimeeew committed:
Repositioned book button

aimeeew committed:
Removed box-sizing - already in bootstrap

aimeeew committed:
Repositioned login button

aimeeew committed:
Resized for larger screen

aimeeew committed:
Centred button titles

aimeeew committed:
Filled pop-outs

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

rmjjjt committed:
Linked Selected From Date With To Start Date.

This stops the user from selecting a date that is before the selected from date.

aimeeew committed:
edited header/footer layout and added logo

aimeeew committed:
positioning and sizing header & logo

aimeeew committed:
edited logo-bar, more

aimeeew committed:
changed appearance of buttons

apgmartin committed:
new time picker

aimeeew committed:
changed button hover effects

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

apgmartin committed:
added CSS to timePicker

rmjjjt committed:
attempted range highlighting

aimeeew committed:
added alien and speech bubble

rmjjjt committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
Renamed main page home

apgmartin committed:
commented on timePicker.js

rmjjjt committed:
integrating calendar to index

rmjjjt committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
Altered hover on date/time boxes

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

benmorrismayden committed:
Add Carpark.php class unit test file

aimeeew committed:
Added mayden logos

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

benmorrismayden committed:
Create unit tests for Carpark class WORKING

ralph-rutter commented on classes/Carpark.php:
```diff
@@ -0,0 +1,52 @@
+ * User: benmorris
+ * Date: 14/11/2016
+ * Time: 14:35
+ */
```
We don't need the auto-generated comments.

ralph-rutter commented on classes/Carpark.php:
```diff
@@ -0,0 +1,52 @@
+
+    public function __construct(PDO $pdo, $name) {
+
+        $statement = $pdo->query('SELECT * FROM `' . self::CARPARK_TABLE_NAME . '` WHERE `name` = \'' . $name . '\';');
```
This query call needs splitting out into prepare and execute, to avoid sql injection vulnerability.

benmorrismayden committed:
Create unit tests for Carpark.php class

benmorrismayden committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ralph-rutter commented on classes/Carpark.php:
```diff
@@ -0,0 +1,52 @@
+    public function __construct(PDO $pdo, $name) {
+
+        $statement = $pdo->query('SELECT * FROM `' . self::CARPARK_TABLE_NAME . '` WHERE `name` = \'' . $name . '\';');
+        $result = $statement->fetch();
```
Passing fetch PDO::FETCH_ASSOC will return just the associative array and not both assoc. and indexed, hence storing less data.

rmjjjt committed:
Added Visitor Calendar Html and JS.

Created a new page for visitor calendar to be loaded from.
Set up JS for visitor calendar to create a new instance and changed headers.

rmjjjt committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ralph-rutter commented on autoload.php:
```diff
@@ -0,0 +1,9 @@
+<?php
+spl_autoload_register('class_autoloader');
+
+function class_autoloader($class) {
```
Needs a docBlock

benmorrismayden committed:
Ammend Carpark.php to use PDOSTatement::execute rather than query and update CarparkTest accordingly

ralph-rutter commented on classes/Carpark.php:
```diff
@@ -0,0 +1,52 @@
+    private $capacity;
+    private $isVisitor;
+
+    public function __construct(PDO $pdo, $name) {
```
needs a DocBlock

apgmartin committed:
added staff date picker

benmorrismayden committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ralph-rutter commented on classes/Carpark.php:
```diff
@@ -0,0 +1,52 @@
+
+    }
+
+    public function getId() {
```
These getters need docBlocks. Also should isVisitor() be getIsVisitor() for consistency?

LukeW20 committed:
updated sql file

apgmartin committed:
merged confilct and button functionality

ralph-rutter commented on js/calendar.js:
```diff
@@ -0,0 +1,74 @@
+$date = new Date()
+$month = $date.getMonth()
+$toStartDate = $date
```
Do you need this variable? It seems only to appear again in line 20, where it still is the same as $date.

apgmartin committed:
added in visitor calendar

LukeW20 committed:
updated sql file

LukeW20 committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ralph-rutter commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+ * User: benmorris
+ * Date: 15/11/2016
+ * Time: 14:17
+ */
```
Don't need automated comments

ralph-rutter replied:
Don't need this comment.

apgmartin committed:
added visitor filter and time animation

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

ross-eastman-mayden commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+<?php
+
+require '/Users/benmorris/Documents/Development/academy/parking-system/classes/Carpark.php';
```
absolute file paths again

ralph-rutter commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+<?php
+
+require '/Users/benmorris/Documents/Development/academy/parking-system/classes/Carpark.php';
+require '/Users/benmorris/Documents/Development/academy/parking-system/vendor/autoload.php';
```
These probably need to be relative paths, so that they'll work for other users.

apgmartin committed:
added some styling to time picker

benmorrismayden committed:
Work so far

benmorrismayden committed:
getBookings function completed

benmorrismayden committed:
Add directory for manual tests

benmorrismayden committed:
Resolve syntax problem in Booking Manager

ralph-rutter commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+            'isVisitor' => 1
+        ]);
+        $mockPDO->method('prepare')->willReturn($mockStatement);
+
```
In order to fetch, the prepared statement needs to be executed after having been prepared, so I would imagine the execute method on $mockStatement will needs to appear somewhere in the test.

ralph-rutter commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+        $mockPDO = $this->createMock('PDO');
+        $mockStatement = $this->createMock('PDOStatement');
+        $mockStatement->method('fetch')->willReturn([]);
+        $mockPDO->method('prepare')->willReturn($mockStatement);
```
The same applies here with the execute method.

ralph-rutter commented on test/classes/CarparkTest.php:
```diff
@@ -0,0 +1,57 @@
+            $carpark = new Carpark($mockPDO, $testName);
+        } catch (Exception $e) {
+            throw new Exception ('Carpark with name ' . $testName . ' does not exist in pdo.');
+        }
```
This test needs some assertions.

benmorrismayden committed:
Writen and tested function for getting bookings between 2 dates

apgmartin committed:
added button and styling to visitor timepicker

ralph-rutter committed:
added copyright info for main logo

ralph-rutter committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

rmjjjt committed:
Added Styling To The Date And Time Sliders.

Deleted main.scss as was a duplicate of main.
Added box styling to match the buttons for the sliding boxes.
Amended some styles for the sliding boxes to fit with adjusted heights - move to px from %.

apgmartin committed:
added visitor filter button and created ajax to availability.php

rmjjjt committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

benmorrismayden committed:
Implement Carpark::getAvailability for staff

benmorrismayden committed:
working

aimeeew committed:
Changed include path to relative

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

rmjjjt committed:
slider color styling

rmjjjt committed:
recompiled styles map

benmorrismayden committed:
Docblocks for Carpark.phpgit status

benmorrismayden committed:
Docblock Carpark.php

benmorrismayden committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

benmorrismayden committed:
Resolve Ralphs comments on CarparkTest.php

apgmartin committed:
added staff and submit buttons and added disabling submission conditions

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
working

apgmartin committed:
put ajax post data in to variables in availability.php

apgmartin committed:
removed var_dumps

apgmartin committed:
created template for availability.php, just needs database stuff

apgmartin committed:
combined js files

apgmartin committed:
removed obsolete files

benmorrismayden committed:
First working version of Carpark::getAvailability()

benmorrismayden committed:
Some input format checking implemented

benmorrismayden committed:
Merge branch 'bookings_function' into display_spaces

benmorrismayden committed:
Copy DbConnector.php from login branch

aimeeew committed:
Added docblocks to getAvailability and check_in_range

apgmartin committed:
sorted out showing availability and made extensible

benmorrismayden committed:
Connect front end to db booking queries

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

apgmartin committed:
connected backend database with availability filter

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

apgmartin committed:
fixed submission bug with staff calendar

apgmartin committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

apgmartin committed:
removed console.logs

apgmartin committed:
changed date convereter to use php native functions

apgmartin committed:
fixed browser caching of button being disabled

apgmartin committed:
added bootstrap js file

apgmartin committed:
changed availability to return array of objects also containing carpark is

apgmartin committed:
added modal popups to confirm bookings

apgmartin committed:
updated modals to confirm exact booking times

apgmartin committed:
formatted modals to look better

apgmartin committed:
changed book button to only view when spaces are available

aimeeew committed:
Working split out availability functions

aimeeew committed:
Docblocked getAvailability method

aimeeew committed:
Abstracted out bookings manager and docblocked getConcurrentBookings

aimeeew committed:
Merge branch 'display_spaces' of github.com:may-den/parking-system into display_spaces

aimeeew committed:
Added docblock for getTimeStampFromTime

aimeeew committed:
Finished docblocks for availability methods

aimeeew committed:
Changed some methods to public for testing and accessibility

aimeeew committed:
Started unit tests for availability functions

Commit:
fix availability dependancies
